xdp/src/tx_loop.rs

```diff
@@ -193,25 +169,27 @@ pub fn tx_loop<T: AsRef<[u8]>, A: AsRef<[SocketAddr]>>(
 
         for (addrs, payload) in batched_items.drain(..) {
             for addr in addrs.as_ref() {
-                // loop until we have space for the next packet
-                loop {
-                    completion.sync(true);
-                    // we haven't written any frames so we only need to sync the consumer position
-                    ring.sync(false);
-
-                    // check if any frames were completed
-                    while let Some(frame_offset) = completion.read() {
-                        umem.release(frame_offset);
+                if ring.available() == 0 || umem.available() == 0 {
```

Comment:
Is the reason we don't just loop on this so that we can avoid calling kick 1 time?

Reply:
correct! kick does a syscall, which when sending Mpps takes an eternity